lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -417,8 +417,27 @@ void FModuleOp::insertPorts(ArrayRef<std::pair<unsigned, PortInfo>> ports) {
   newAnnos.reserve(newNumArgs);
   newSyms.reserve(newNumArgs);
 
-  auto EmptyArray = ArrayAttr::get(getContext(), {});
-  bool addAnnos = existingAnnos.size() > 0;
+  auto emptyArray = ArrayAttr::get(getContext(), {});
+  auto emptyString = StringAttr::get(getContext(), "");
+  bool addAnnos =
+      llvm::any_of(existingAnnos,
+                   [](auto attr) {
+                     return attr &&
```

Comment:
do you need to check `attr` here?

Reply:
can you just check if the `existingAnnos` array is empty like before? I think `getPortAnnotations()` is still returning an empty array.

Reply:
My thought here was that this gives it a chance to clean up an array that is full of empty anno lists or full of empty symbols. But may e that's not needes?

Reply:
it would be nice to canonicalize arrays of empties to an empty array.

Reply:
Leaving this in for the sake of canonicalization.